### PR TITLE
Robust segvit compatibility + new dataset.

### PR DESCRIPTION
**Pull request by ekellbuch**
This commit makes the code in robust_segvit compatible externally and adds config files to experiment with a benchmark for open set recognition in image segmentation, namely the street hazards dataset from https://arxiv.org/abs/1911.11132.

### CODE REVIEW

File: experimental/robust_segvit/metrics_multihost.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  tp = jnp.sum(all_tp, 0)
+  fp = jnp.sum(all_fp, 0)
+  tn = jnp.sum(all_tn, 0)
+  fn = jnp.sum(all_fn, 0)
+
+  return tp, fp, tn, fn
+
+
+class ComputeAUCMetric:
+  """Calculate auc metrics across multiple hosts."""
```
Comment (dusenberrymw) [inline experimental/robust_segvit/metrics_multihost.py:52]: We could probably replace all of the custom multihost AUC metric code we wrote with the new `host_all_gather_metrics` function we added, using it in the same way as for ECE. Fine not to do it now, but could be a nice cleanup.

File: experimental/robust_segvit/pretrainer_utils.py [hunk 2]
```diff
@@ -200,3 +201,58 @@ def convert_torch_to_jax_checkpoint(
...
+  restored_params = dict(restored_params)
+
+  train_state = train_utils.TrainState()
+  # pytype: disable=wrong-arg-types
+  restored_train_state = train_state.replace(  # pytype: disable=attribute-error
+      optimizer={"target": restored_params},)
+  # pytype: enable=wrong-arg-types
+
+  # free memory
+  del restored_params
```
Comment (dusenberrymw) [inline experimental/robust_segvit/pretrainer_utils.py:257]: This will be deleted anyway when the function returns on the next line, so I don't think we need to explicitly delete it here.

File: experimental/robust_segvit/uncertainty_metrics.py [hunk 3]
```diff
@@ -118,29 +117,43 @@ def get_uncertainty_confusion_matrix(
...
-  entropy = get_entropy_from_logits(logits)
+  # A given patch is accurate if its acc > accuracy_threshold
+  binary_acc_map = reduce_2dmap_weighted(accuracy_map,
+                                         weights,
+                                         window_size=window_size,
+                                          threshold=accuracy_th).astype(jnp.float32)
+
+  # Calculate uncertainty map:
+  if uncertainty_measure == 'softmax':
+    uncertainty_map = jnp.max(jnp.exp(logits) / jnp.sum(jnp.exp(logits), -1, keepdims=True), -1)
```
Comment (dusenberrymw) [inline experimental/robust_segvit/uncertainty_metrics.py:140]: could use `jax.nn.softmax`

File: experimental/robust_segvit/uncertainty_metrics_test.py [hunk 4]
```diff
@@ -1,92 +0,0 @@
...
-"""Tests for uncertainty_metrics."""
-
-from absl.testing import absltest
-from absl.testing import parameterized
-import jax.numpy as jnp
-from uncertainty_metrics import reduce_2dmap  # local file import from experimental.robust_segvit
-from uncertainty_metrics import SegmentationUncertaintyMetrics  # local file import from experimental.robust_segvit
-
-
-class UncertaintyMetricsTest(parameterized.TestCase):
```
Comment (dusenberrymw) [inline experimental/robust_segvit/uncertainty_metrics_test.py:25]: What do you think about updating these tests instead of deleting them?

**Review by dusenberrymw — CHANGES_REQUESTED** [review]

File: experimental/robust_segvit/custom_segmentation_trainer.py [hunk 5]
```diff
@@ -160,17 +174,51 @@ def evaluate(train_state: train_utils.TrainState,
...
     eval_batch = next(dataset.valid_iter)
     e_batch, e_logits, e_metrics, confusion_matrix, unc_confusion_matrix = eval_step_pmapped(
         train_state=train_state, batch=eval_batch)
     eval_metrics.append(train_utils.unreplicate_and_get(e_metrics))
+
+    probs = jax.nn.softmax(e_logits, axis=-1)
+
+    # TODO(kellybuchanan): add masking to ece metric in rm.
+    # updates on each host separately
+    ece_metric.update_state(e_batch['label'], probs, sample_weight=e_batch['batch_mask'])
```
Comment (dusenberrymw) [inline experimental/robust_segvit/custom_segmentation_trainer.py:204]: Based on the comment in https://github.com/google-research/robustness_metrics/pull/77, can we move the binary masking logic here such that we pass in a masked subset of labels and probs as needed without changing the ECE metric API?

**Review by dusenberrymw — CHANGES_REQUESTED** [review]

File: experimental/robust_segvit/uncertainty_metrics.py (hunk 3 above)
Comment (ekellbuch) [reply experimental/robust_segvit/uncertainty_metrics.py]: Done.

File: experimental/robust_segvit/pretrainer_utils.py (hunk 2 above)
Comment (ekellbuch) [reply experimental/robust_segvit/pretrainer_utils.py:255]: Done

File: experimental/robust_segvit/metrics_multihost.py (hunk 1 above)
Comment (ekellbuch) [reply experimental/robust_segvit/metrics_multihost.py:73]: Done
